sympy/calculus/finite_diff.py

```diff
@@ -167,6 +167,9 @@ def finite_diff_weights(order, x_list, x0=S.One):
 
     """
     # The notation below closely corresponds to the one used in the paper.
+    order = S(order)
+    if not order.is_number:
+        raise ValueError("Cannot handle symbolic order.")
```

Comment:
Can these three conditions be combined? After simpification the result should be a non-negative Integer so:
```python
if not order.is_Integer and order >= 0:
     raise ValueError("Order must be an explicit nonnegative integer.")
```

Reply:
Yes, for trading clarity in error message.